### PR TITLE
#44: Create Primary Layout

### PR DESCRIPTION
**Pull request by DominikWas**
Completes ticket #44 
----------------------
app.component.html
* Added a consistent toolbar for the app, each component hopefully could fit under the main toolbar

app.component.ts
* The main title of the framework has been changed to'OmegaSys' for now meaning 'Omega System' for 'Project Omega', name can be changed in the future

app.module.ts
* Imported various modules, mainly material/forms (design uses) and routing for navigation, 2 new components added with comments for organisation

index.html
* Changed the title to represent the framework's name, 'OmegaSys' for now, background addition

styles.scss
* Added basic universal styling for consistent design

package.json
* Added support for cdk (useful for material usage), material (for design neat design implementations) and a mock JSON server api that will be used for mocking JSON data/server relationship locally

package-lock.json
* Added extra dependencies (see above package.json description please)

customers.component.html/scss/spec.ts/ts
* Sample customer component added, empty shell for now

dashboard.component.html
* Dashboard that supports a router link to customer component
* Style addition with attributes (e.g.: 'rowHeight', 'gutterSize'
* Added tiles for future navigation to the planned pages, 'Customers' page link works

dashboard.component.scss
* Light blue styling for tiles

dashboard.component.ts
* Created a component for the dashboard

dashboard.component.spec.ts
* Default test - Truthy test

app-routing.module.ts
* Added a basic routing mechanism for navigation - main dashboard (expected to be shown after authentication page) added, customer component added

### CODE REVIEW

**Comment by HarryPi** [comment]
Hi Dom,
A couple of points
- I know this is a very simple pull request but it would be good to make 1-2 quick and easy tests  i.e to test the redirect. However one can argue that these tests are unnecessary and waste of time therefor i wont argue on that it would just be a good start to get them setup dependencies etc. Its easier to understand Angular testing when things are simple.
- Consider Lazy Loading / Pre-Loading all your routing modules considering this will be a big application? ( Again not a first priority however good to keep in mind). This will allow us to load the chunks of the application as needed and not all at once.
- From personal experience i find mat-grid-list to be a bit of clunky and hard to handle i would suggest looking at https://github.com/angular/flex-layout/wiki/fxFlex-API it utilities the new CSS Flex box and made as container for angular by the angular team. Again this is personal preference.

Beyond the points that are all up for discussion / preference i do not see anything wrong with the pull request.

**Comment by DominikWas** [comment]
Wicked, I’ll add a few more tests soon. Ahh I was looking for that @ flexbox! And for sure, pre-loading can be added. Thanks Harry!

File: AdminClient/src/app/dashboard/dashboard.component.scss [hunk 1]
```diff
@@ -0,0 +1,4 @@
+mat-grid-tile {
```
Comment (markscamilleri) [inline AdminClient/src/app/dashboard/dashboard.component.scss]: Should this be an element selector or class selector?

Comment (DominikWas) [reply AdminClient/src/app/dashboard/dashboard.component.scss]: Made this for element selectors, just for the dashboard tiles - didn't think I'd use other tiles for dashboard. Can modularise it more if needed, let me know what you think!

**Comment by markscamilleri** [comment]
Blocked by #42

**Comment by markscamilleri** [comment]
Fixed some visual stuff - but @DominikWas @HarryPi  I don't know how to add the actual menu drawer. I think this should be added asap since we need it to navigate anywhere else.

**Comment by HarryPi** [comment]
> Fixed some visual stuff - but @DominikWas @HarryPi I don't know how to add the actual menu drawer. I think this should be added asap since we need it to navigate anywhere else.

Do you mean the Sidenav?

**Comment by HarryPi** [comment]
Also i am checking this branch so why does this branch has a store for Auth and a reducer when its not using NgRx ? @markscamilleri @DominikWas

**Comment by markscamilleri** [comment]
> > Fixed some visual stuff - but @DominikWas @HarryPi I don't know how to add the actual menu drawer. I think this should be added asap since we need it to navigate anywhere else.
> 
> Do you mean the Sidenav?

Yes, yes I do

**Comment by markscamilleri** [comment]
> Also i am checking this branch so why does this branch has a store for Auth and a reducer when its not using NgRx ? @markscamilleri @DominikWas Furthermore; Auth is a Page? a page is an ionic component not a native angular component.

1. Ngrx was added cause it wasn't building (it was being used in another file which will be edited and used in #45)

2. I'll change auth to . component in #45

It's either that or I can remove Auth from this one, what do you prefer?

**Comment by DominikWas** [comment]
I'll open a separate issue for side nav, any other changes to this will come as a separate pull request/issue.

**Comment by HarryPi** [comment]
> I'll open a separate issue for side nav, any other changes to this will come as a separate pull request/issue.

Hi dom see #68 I have already made a branch and will take care of this in the mean time take care of the other issues of the other pull request i believe.

**Comment by DominikWas** [comment]
Just noticed #68,  alright thank you

**Pull request merged**